### ISSUE TITLE
Unable to include custom java source files

### ISSUE THREAD

**Issue by kadaj** [technomancy/leiningen#2228]
When I add the `:java-source-paths ["src/java"]` in the `projects.clj` `lein repl` command fails with the error
```
Java compiler not found; Be sure to use java from a JDK
rather than a JRE by modifying PATH or setting JAVA_CMD.
```
I have JDK in the path. `javac -version` gives `javac 1.8.0_91`. It works when the java-source-paths is not added. Tried in Windows 7, Leiningen 2.7.1 on Java 1.8.0_91 Java HotSpot(TM) 64-Bit Server VM.  No issues with Ubuntu.

**Comment by kadaj** [comment]
I am having problem with Ubuntu as well. I loads the repl but error is thrown at the beginning. The clojure code in unable to find the java classes. For testing I tried to run the [clj-antlr demo](https://github.com/aphyr/clj-antlr/tree/master/demo) project.
```
(#error {
 :cause Unable to resolve symbol: visitor in this context
 :via
 [{:type clojure.lang.Compiler$CompilerException
   :message java.lang.RuntimeException: Unable to resolve symbol: visitor in this context, compiling:(clj_antlr_demo/edn.clj:59:3)
   :at [clojure.lang.Compiler analyze Compiler.java 6688]}
  {:type java.lang.RuntimeException
   :message Unable to resolve symbol: visitor in this context
   :at [clojure.lang.Util runtimeException Util.java 221]}]
 :trace
 [[clojure.lang.Util runtimeException Util.java 221]
  [clojure.lang.Compiler resolveIn Compiler.java 7164]
  [clojure.lang.Compiler resolve Compiler.java 7108]
  [clojure.lang.Compiler analyzeSymbol Compiler.java 7069]
  [clojure.lang.Compiler analyze Compiler.java 6648]
  [clojure.lang.Compiler analyze Compiler.java 6625]
  [clojure.lang.Compiler$InvokeExpr parse Compiler.java 3766]
  [clojure.lang.Compiler analyzeSeq Compiler.java 6870]
  [clojure.lang.Compiler analyze Compiler.java 6669]
  [clojure.lang.Compiler access$300 Compiler.java 38]
  [clojure.lang.Compiler$DefExpr$Parser parse Compiler.java 589]
  [clojure.lang.Compiler analyzeSeq Compiler.java 6868]
  [clojure.lang.Compiler analyze Compiler.java 6669]
  [clojure.lang.Compiler analyze Compiler.java 6625]
  [clojure.lang.Compiler eval Compiler.java 6931]
  [clojure.lang.Compiler load Compiler.java 7379]
  [clojure.lang.RT loadResourceScript RT.java 372]
  [clojure.lang.RT loadResourceScript RT.java 363]
  [clojure.lang.RT load RT.java 453]
  [clojure.lang.RT load RT.java 419]
  [clojure.core$load$fn__5677 invoke core.clj 5893]
  [clojure.core$load invokeStatic core.clj 5892]
  [clojure.core$load doInvoke core.clj 5876]
  [clojure.lang.RestFn invoke RestFn.java 408]
  [clojure.core$load_one invokeStatic core.clj 5697]
  [clojure.core$load_one invoke core.clj 5692]
  [clojure.core$load_lib$fn__5626 invoke core.clj 5737]
  [clojure.core$load_lib invokeStatic core.clj 5736]
  [clojure.core$load_lib doInvoke core.clj 5717]
  [clojure.lang.RestFn applyTo RestFn.java 142]
  [clojure.core$apply invokeStatic core.clj 648]
  [clojure.core$load_libs invokeStatic core.clj 5774]
  [clojure.core$load_libs doInvoke core.clj 5758]
  [clojure.lang.RestFn applyTo RestFn.java 137]
  [clojure.core$apply invokeStatic core.clj 650]
  [clojure.core$use invokeStatic core.clj 5860]
  [clojure.core$use doInvoke core.clj 5860]
  [clojure.lang.RestFn invoke RestFn.java 436]
  [clj_antlr_demo.bin$eval14$loading__5569__auto____15 invoke bin.clj 1]
  [clj_antlr_demo.bin$eval14 invokeStatic bin.clj 1]
  [clj_antlr_demo.bin$eval14 invoke bin.clj 1]
  [clojure.lang.Compiler eval Compiler.java 6927]
  [clojure.lang.Compiler eval Compiler.java 6916]
  [clojure.lang.Compiler load Compiler.java 7379]
  [clojure.lang.RT loadResourceScript RT.java 372]
  [clojure.lang.RT loadResourceScript RT.java 363]
  [clojure.lang.RT load RT.java 453]
  [clojure.lang.RT load RT.java 419]
  [clojure.core$load$fn__5677 invoke core.clj 5893]
  [clojure.core$load invokeStatic core.clj 5892]
  [clojure.core$load doInvoke core.clj 5876]
  [clojure.lang.RestFn invoke RestFn.java 408]
  [clojure.core$load_one invokeStatic core.clj 5697]
  [clojure.core$load_one invoke core.clj 5692]
  [clojure.core$load_lib$fn__5626 invoke core.clj 5737]
  [clojure.core$load_lib invokeStatic core.clj 5736]
  [clojure.core$load_lib doInvoke core.clj 5717]
  [clojure.lang.RestFn applyTo RestFn.java 142]
  [clojure.core$apply invokeStatic core.clj 648]
  [clojure.core$load_libs invokeStatic core.clj 5774]
  [clojure.core$load_libs doInvoke core.clj 5758]
  [clojure.lang.RestFn applyTo RestFn.java 137]
  [clojure.core$apply invokeStatic core.clj 648]
  [clojure.core$require invokeStatic core.clj 5796]
  [clojure.core$require doInvoke core.clj 5796]
  [clojure.lang.RestFn invoke RestFn.java 408]
  [user$eval5 invokeStatic form-init1440287508227871770.clj 1]
  [user$eval5 invoke form-init1440287508227871770.clj 1]
  [clojure.lang.Compiler eval Compiler.java 6927]
  [clojure.lang.Compiler eval Compiler.java 6916]
  [clojure.lang.Compiler eval Compiler.java 6916]
  [clojure.lang.Compiler load Compiler.java 7379]
  [clojure.lang.Compiler loadFile Compiler.java 7317]
  [clojure.main$load_script invokeStatic main.clj 275]
  [clojure.main$init_opt invokeStatic main.clj 277]
  [clojure.main$init_opt invoke main.clj 277]
  [clojure.main$initialize invokeStatic main.clj 308]
  [clojure.main$null_opt invokeStatic main.clj 342]
  [clojure.main$null_opt invoke main.clj 339]
  [clojure.main$main invokeStatic main.clj 421]
  [clojure.main$main doInvoke main.clj 384]
  [clojure.lang.RestFn invoke RestFn.java 421]
  [clojure.lang.Var invoke Var.java 383]
  [clojure.lang.AFn applyToHelper AFn.java 156]
  [clojure.lang.Var applyTo Var.java 700]
  [clojure.main main main.java 37]]}
nREPL server started on port 38494 on host 127.0.0.1 - nrepl://127.0.0.1:38494
REPL-y 0.3.7, nREPL 0.2.12
Clojure 1.8.0
Java HotSpot(TM) 64-Bit Server VM 1.8.0_112-b15
    Docs: (doc function-name-here)
          (find-doc "part-of-name-here")
  Source: (source function-name-here)
 Javadoc: (javadoc java-object-or-class-here)
    Exit: Control+D or (exit) or (quit)
 Results: Stored in vars *1, *2, *3, an exception in *e

clj-antlr-demo.bin=>
```

**Comment by danielcompton** [comment]
What happens if you explicitly set the JAVA_CMD environment variable?

**Comment by kadaj** [comment]
It works when JAVA_CMD is set. Thanks.

**Comment by danielcompton** [comment]
Is your path exported? It looks like there may still be a bug here.

**Comment by kadaj** [comment]
Yes JAVA_CMD is added to env variables. I have JAVA_HOME and in path \bin added so that cmd can find javac etc.

**Comment by euglv** [comment]
What should be value of `JAVA_CMD` ?

**Comment by jimmyhmiller** [comment]
For people who google this and land here. 

I had no `JAVA_CMD` setup. When I ran `which java` it returned 

```bash
/Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java
```

That is java from a JRE. I do have the jdk installed. To find out where I ran `which javac`. That returned

```bash
/usr/bin/javac
```

So it turns out I also have a `/usr/bin/java`. I ran this and everything worked fine.

```bash
export JAVA_CMD=/usr/bin/java
```

Hopefully this helps people in the future.
